README.md

```diff
@@ -1,58 +1,164 @@
+- They are a language for writing schemas for your data.
+- They define a binary format for serializing your data.
+
+These two independent traits functions work together to allow your project and everyone who interacts with it to define messages, fields, and service APIs in the exact same way.   In a practical sense as it relates to **Protobuf-ES**, this means no more disparate JSON types all over the place.  Instead, you define a common schema in a Protobuf file, such as:
+
+```proto
+message Foo {
+   string bar = 1;
+   boolean baz = 2;
+   int32 bing = 3;
 }
 ```
 
-Is compiled to an ECMAScript class that can be used like this:
+And now your schema can be used with a consistent approach throughout your code.  The benefits can extend to any application that interacts with yours as well.  This is because the Protobuf file can be used to generate types in many languages.  So, now all applications can use the same definition of `Foo` as defined above.  And when everyone agrees on what a `Foo` is, communication occurs much more easily.  The added bonus is that no one has to write any boilerplate code to make this happen.  [Code generators](https://www.npmjs.com/package/@bufbuild/protoc-gen-es) handle all of this for you.
+
+Protocol Buffers also allow you to serialize this structured data.  So, your application running in the browser can send a `Foo` object to a backend running an entirely different language, but using the exact same definition.  Using an RPC framework like [Connect-Web](https://github.com/bufbuild/connect-web), your data is serialized into bytes on the wire and then deserialized at its destination using the defined schema.
+
+## Quickstart
+
+To get started generating code right away, first make sure you have [Buf](https://docs.buf.build/installation) installed.  If desired, you can also use `protoc`.  For full instructions on generating code, visit the [docs](https://github.com/bufbuild/protobuf-es/blob/main/docs/generated_code.md) for the `protoc-gen-es` plugin.
+
+### Installation
+
+Install the necessary packages.  For details on each package, see [below](#packages).  The statement listed below is the most likely one you will need.
+
+```bash
+npm install @bufbuild/protobuf
+npm install --save-dev @bufbuild/protoc-gen-es
+```
+
+### Code Generation
+
+Once installed, download the example file [example.proto](https://raw.githubusercontent.com/bufbuild/protobuf-es/main/packages/protobuf-test/extra/example.proto) and place it into a `protos/` directory.
+
+Next, create a `buf.gen.yaml` file in your project root that looks like this:
+
+```yaml
+# Learn more: https://docs.buf.build/configuration/v1/buf-gen-yaml
+version: v1
+plugins:
+  - name: es
+    path: ./node_modules/.bin/protoc-gen-es
+    opt: target=ts
+    out: src/gen
+```
+
+Finally, run `buf generate`.  You should now see a generated file at `src/gen/example_pb.ts` that contains a class named `User`.  From here, you can begin to work with your schema.
```

Comment:
The quickstart [here](https://github.com/timostamm/protobuf-ts#quickstart) takes only three steps to generate code, and seems a bit easier to follow, TBH.

If we already had the buf CLI available through npm, it could be:

> 1. Install the buf CLI, the code generator and the runtime library:
> 
> ```bash
> npm install @bufbuild/buf @bufbuild/protobuf @bufbuild/protoc-gen-es
> ```
> 
> 2. Create a `buf.gen.yaml` file:
> 
> ```yaml
> # Learn more: https://docs.buf.build/configuration/v1/buf-gen-yaml
> version: v1
> plugins:
>   - name: es
>     opt: target=ts
>     out: src/gen
> ```
> 
> 3. Generate [person.proto](https://github.com/bufbuild/protobuf-es/blob/main/packages/protobuf-test/extra/example.proto):
> 
> ```bash
> npx buf generate https://github.com/bufbuild/protobuf-es.git#subdir=packages/protobuf-test/extra -- path=person.proto
> ```

Or, if generating from git is too obscure:

> 1. Install the buf CLI, the code generator and the runtime library:
> 
> ```bash
> npm install @bufbuild/buf @bufbuild/protobuf @bufbuild/protoc-gen-es
> ```
> 
> 2. Create a `buf.gen.yaml` file:
> 
> ```yaml
> # Learn more: https://docs.buf.build/configuration/v1/buf-gen-yaml
> version: v1
> plugins:
>   - name: es
>     opt: target=ts
>     out: src/gen
> ```
> 
> 3. Download [person.proto](https://github.com/bufbuild/protobuf-es/blob/main/packages/protobuf-test/extra/example.proto) into a `proto/` directory
> ```bash
> mkdir proto
> curl https://raw.githubusercontent.com/bufbuild/protobuf-es/main/packages/protobuf-test/extra/person.proto > proto/person.proto
> ```
> 
> 4. Generate
> ```bash
> buf generate proto
> ```

Any chance we can get closer to this?

Reply:
Modified to use your second example (without generating from git)